database/bfgd/postgres/postgres.go

```diff
@@ -946,64 +927,40 @@ func (p *pgdb) AccessPublicKeyExists(ctx context.Context, publicKey *bfgd.Access
 	log.Tracef("AccessPublicKeyExists")
 	defer log.Tracef("AccessPublicKeyExists exit")
 
-	const sql = `
+	const q = `
 		SELECT EXISTS (
 			SELECT * FROM access_public_keys WHERE public_key = $1
 		)
 	`
 
-	rows, err := p.db.QueryContext(ctx, sql, publicKey.PublicKey)
-	if err != nil {
+	var exists bool
+	if err := p.db.QueryRowContext(ctx, q, publicKey.PublicKey).Scan(&exists); err != nil {
 		return false, err
 	}
 
-	defer rows.Close()
-
-	for rows.Next() {
-		var exists bool
-		err = rows.Scan(&exists)
-		if err != nil {
-			return false, err
-		}
-
-		return exists, nil
-	}
-
-	if err = rows.Err(); err != nil {
-		return false, err
-	}
-
-	return false, errors.New("should not get here")
+	return exists, nil
 }
 
 func (p *pgdb) AccessPublicKeyDelete(ctx context.Context, publicKey *bfgd.AccessPublicKey) error {
 	log.Tracef("AccessPublicKeyDelete")
-	log.Tracef("AccessPublicKeyDelete exit")
+	defer log.Tracef("AccessPublicKeyDelete exit")
 
-	sql := fmt.Sprintf(`
-		WITH deleted AS (
-			DELETE FROM access_public_keys WHERE public_key = $1
-			RETURNING *
-		) SELECT count(*) FROM deleted;
-	`)
+	const q = `
+		DELETE FROM access_public_keys WHERE public_key = $1
+	`
 
-	rows, err := p.db.QueryContext(ctx, sql, publicKey.PublicKey)
+	res, err := p.db.ExecContext(ctx, q, publicKey.PublicKey)
 	if err != nil {
 		return err
 	}
 
-	for rows.Next() {
-		var count int
-		if err := rows.Scan(&count); err != nil {
-			return err
-		}
-
-		return database.NotFoundError("public key not found")
-	}
```

Comment:
This behaviour is rather confusing, but I believe it is supposed to return a `database.NotFoundError` if nothing was deleted? The refactored code should have the proper behaviour, but this appears to always return a `database.NotFoundError`?

Reply:
yes I think you're right @joshuasing ; we were missing a check against the count 🤦